### PR TITLE
reset the HOTP counter

### PR DESCRIPTION
**Pull request by cornelinux**
If the slot 0x10 is used earlier the OTP counter
can be at any higher number.
So we reset the OTP counter to 0, so that the
tests according to RFC4226 will succeed.

### CODE REVIEW

File: unittest/test_HOTP.cc [hunk 1]
```diff
@@ -58,14 +58,25 @@ TEST_CASE("Test HOTP codes according to RFC", "[HOTP]") {
     //strcpy(reinterpret_cast<char *>(hwrite.slot_secret), "");
     const char* secretHex = "3132333435363738393031323334353637383930";
     hexStringToByte(hwrite.slot_secret, secretHex);
-    //hwrite.slot_config; //TODO check various configs in separate test cases
+
+    // We need to reset the counter of the slot, in case the slot was used earlier
+    hwrite.slot_counter[0] = 0;
```
Comment (szszszsz) [inline unittest/test_HOTP.cc]: Hi! Please use instead: 

> memset (hwrite.slot_counter,0,8);

File: README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Build
```
Comment (szszszsz) [inline README.md]: Hi!
I have already added readme to master but have not rebased this branch yet. Could you remove the readme commit and modify current readme instead?

File: README.md [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+To build the lib on ubuntu 14.04LTS you need to
+
+1. Install
+
+  * catch
+
+2. Run
+
+   mkdir build
+   make CXXFLAGS="-std=c++14 -fPIC -stdlib=libc++" all
```
Comment (szszszsz) [inline README.md]: Hi! Why are these needed?

> CXXFLAGS="-std=c++14 -fPIC -stdlib=libc++"

File: unittest/test_HOTP.cc (hunk 1 above)
Comment (cornelinux) [inline unittest/test_HOTP.cc]: Thanks a lot. My C/C++ knowlegde is a bit limited ;-)

File: README.md (hunk 3 above)
Comment (cornelinux) [inline README.md]: On my ubuntu it will fail otherwise.

**Comment by cornelinux** [comment]
I will rebase the init with memset

**Comment by szszszsz** [comment]
Looks great! I would only ask to squash all this commits into the one by running:
`git rebase -i`
 and in editor setting all commits besides first to squash (s), then
`git push -f`
Let me know in case of any problems.

**Comment by cornelinux** [comment]
I somehow messed up this branch.
I can not simply rebase it since I already pushed it. :-/
I will create a new pull request.
